polygeist/tools/cgeist/Lib/ValueCategory.cc

```diff
@@ -55,6 +56,19 @@ mlir::Value ValueCategory::getValue(mlir::OpBuilder &builder) const {
 void ValueCategory::store(mlir::OpBuilder &builder, mlir::Value toStore) const {
   assert(isReference && "must be a reference");
   assert(val && "expect not-null");
+  if (toStore.getType().isInteger(1)) {
+    // Ad-hoc extension of booleans
+    const auto GetElementType = [](auto Ty) -> Type {
+      return Ty.getElementType();
+    };
+    auto ElementType =
+        llvm::TypeSwitch<Type, Type>(val.getType())
+            .Case<LLVM::LLVMPointerType>(GetElementType)
+            .Case<MemRefType>(GetElementType)
+            .Default([](auto) -> Type { llvm_unreachable("Unhandled type"); });
+    toStore = builder.createOrFold<arith::ExtUIOp>(builder.getUnknownLoc(),
+                                                   ElementType, toStore);
+  }
```

Comment:
As we're gradually introducing the `i1` type in cgeist, we will find issues as this one. We might want to store `i1` values in `i8` references. We handle it here. (`ExtUIOp` as we treat Boolean values as unsigned by default for now).